contracts/contracts/system_contracts/kaiabridge/Operator.sol

```diff
@@ -0,0 +1,488 @@
+
+        // Fill the first index(0) with dummy tx
+        addTransaction(address(0), "", 0);
+
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+    }
+
+    function _authorizeUpgrade(address newImplementation) internal virtual override onlyOwner {}
+
+    function supportsInterface(bytes4 interfaceId) external override pure returns (bool) {
+        return interfaceId == type(IOperator).interfaceId;
+    }
+
+    /// @dev See {IOperator-addOperator}
+    function addOperator(address operator)
+        public
+        override
+        onlyGuardian
+        operatorDoesNotExist(operator)
+        notNull(operator)
+    {
+        isOperator[operator] = true;
+        operators.push(operator);
+        emit OperatorAddition(operator);
+    }
+
+    /// @dev See {IOperator-removeOperator}
+    function removeOperator(address operator)
+        public
+        override
+        onlyGuardian
+        operatorExists(operator)
+    {
+        isOperator[operator] = false;
+        for (uint64 i=0; i<operators.length - 1; i++)
+            if (operators[i] == operator) {
+                operators[i] = operators[operators.length - 1];
+                break;
+            }
+        operators.pop();
+        if (minOperatorRequiredConfirm > operators.length) {
+            changeRequirement(uint8(operators.length));
+        }
+        emit OperatorRemoval(operator);
+    }
+
+    /// @dev See {IOperator-replaceOperator}
+    function replaceOperator(address operator, address newOperator)
+        public
+        override
+        onlyGuardian
+        operatorExists(operator)
+        operatorDoesNotExist(newOperator)
+    {
+        for (uint64 i=0; i<operators.length; i++) {
+            if (operators[i] == operator) {
+                operators[i] = newOperator;
+                break;
+            }
+        }
+        isOperator[operator] = false;
+        isOperator[newOperator] = true;
+        emit OperatorRemoval(operator);
+        emit OperatorAddition(newOperator);
+    }
+
+    /// @dev See {IOperator-changeGuardian}
+    function changeGuardian(address newGuardian)
+        public
+        override
+        onlyGuardian
+    {
+        emit ChangeGuardian(guardian, newGuardian);
+        guardian = newGuardian;
+    }
+
+    /// @dev See {IOperator-changeBridge}
+    function changeBridge(address newBridge)
+        public
+        override
+        onlyGuardian
+    {
+        emit ChangeBridge(bridge, newBridge);
+        bridge = newBridge;
+    }
+
+    /// @dev See {IOperator-changeRequirement}
+    function changeRequirement(uint8 _minOperatorRequiredConfirm)
+        public
+        override
+        onlyGuardian
+        validRequirement(operators.length, _minOperatorRequiredConfirm)
+    {
+        minOperatorRequiredConfirm = _minOperatorRequiredConfirm;
+        emit RequirementChange(_minOperatorRequiredConfirm);
+    }
+
+    function submitTransaction(address to, bytes calldata data, uint256 uniqUserTxIndex)
```

Comment:
Is it ok to allow `uniqUserTxIndex == 0`? I don't get the point of `submission2TxID`. If it's for searching the txId by `uniqUserTxIndex` for the user, then why don't we have something like `mapping(address => uint256) user2UniqIdx`?

Reply:
The purpose of this storage is the correct one you found. If this explicit conversion is not necessary, the parameter value should be zero.